src/libfetchers/github.cc

```diff
@@ -280,6 +280,15 @@ struct GitArchiveInputScheme : InputScheme
         return {accessor, input};
     }
 
+    bool isLocked(const Input & input) const override
+    {
+        /* Since we can't verify the integrity of the tarball from the
+           Git revision alone, we also require a NAR hash for
+           locking. FIXME: in the future, we may want to require a Git
+           tree hash instead of a NAR hash. */
+        return input.getRev().has_value() && input.getNarHash().has_value();
+    }
+
```

Comment:
This is quite a breaking change (and it also breaks the `githubFlakes` test, although that was hidden behind other breakages)

Reply:
(not to say that we necessarily shouldn't have it, but it should at the very least get a release note)